Kajabi/sage-lib
# Label and Dropdown Updates

## Description

This PR adds some new features to Label and SelectDropdown to ease composition for common patterns:

- Adds `LabelSecondaryButton` that encapsulates settings that need to be used consistently for the `secondaryButton` option in `Label`s.
- Adds a `LabelGroup` component for display inline/wrapping lists of `Label`s.
- Adds an option on `SelectDropdown` for showing selected items as `Label`s below the dropdown that can also be removed (React only)

### Screenshots

New `showSelectionsAsLabels` option on React SelectDropdown:

![Screen Shot 2021-02-10 at 10 41 49 AM](https://user-images.githubusercontent.com/17955295/107533215-b4edaa00-6b8c-11eb-911a-2201eeead554.png)

No visual changes otherwise.

Bonus: Adds `fullWidth` prop to React Button to sync up with the one in Rails. 

## Test notes

See Elements > Label and Elements > Dropdown to verify no visual changes or errors.
See Storybook > Dropdown > Multiselect to see new component in use.

### Steps for testing

These are new features that do not create issues with existing use of these components but do open up opportunities to simplify some existing implementations.

Verify existing uses are unaffected in the following:

1. Label: ensure labels look same and function as they have to date:

    - [ ] Product outline editor (React): publish status dropdowns
    - [ ] Contact import bulk actions step (React): Tag selections that appear below dropdown
    - [ ] Settings > Domain (Rails): see `Kajabi default` is displaying as before
    - [ ] Offers (Rails): publish status label appears as expected (FYI: clicking off of this dropdown does not appear to function as expected and dismiss the dropdown. This appears to be the functionality before this PR as well. A patch may be needed to how this component is implemented here.)

1. Dropdown:
    - [ ] Product outline editor (React): any existing dropdowns
    - [ ] Contact import bulk actions step (React): Tag selection dropdown
    - [ ] Offers (Rails): ... options menu
    - [ ] Product catalogue (Rails): any dropdowns

BONUS: Button: no changes should affect React-based buttons. Confirm that any buttons in the follow appear as expected:
  - [ ] New Contact import process
  - [ ] Product outline manager
